jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java

```diff
@@ -45,12 +36,9 @@
  *
  * <p>The cookie handling provided by this class is guided by the Servlet specification and RFC6265.
  */
-public class HttpFields implements Iterable<HttpField>
+public class HttpFields implements Iterable<HttpField>, HttpFieldList
```

Comment:
We are limited in how much we can change HttpFields because it is part of some public interfaces.   This change make `HttpFields` a builder for the `HttpFields.Immutable` inner class.    It's not what I'd do if green fields.     Note that for `HttpURI` I have made it truly immutable and added `HttpURI.Builder`, which I think is a better way to go... I did that differently because that is more of an internal class.

I'm not sure what you mean by "internal interface"?  There is an external `HttpEventList` interface that is implemented by both `HttpFields` and `HttpFields.Immutable`